### PR TITLE
[SCRUM-175] Add loading & error states to MetaMaskNotificationModal component

### PR DESCRIPTION
**Pull request by Anaphase**
This PR adds three new states to the MetaMaskNotificationModal:

1. Shown after the MM transaction popup appears: "Please wait while your transaction is sent by MetaMask..."
2. Shown after MM resolves the contract call promise (after TX receipt is available I think?): "Your transaction has been submitted!"
3. An error state that shows if MM returns an error

As a fun bonus, this PR also adds `clearModal()` methods to both stake modals so their inputs are after the modal is dismissed.

### CODE REVIEW

File: src/components/modals/StakeTokensModal.vue [hunk 1]
```diff
@@ -6,20 +6,23 @@
...
+      label-size="sm"
+      label-for="stakeAmount"
+      label="Number of tokens to stake"
     >
       <b-form-input
         required
         id="stakeAmount"
-        type="text"
+        ref="stakeAmount"
+        type="number"
```
[the remark this answers is not in the document]
Comment (Anaphase) [reply src/components/modals/StakeTokensModal.vue]: I definitely wouldn't use `type="tel"`, as that displays a phone number pad on mobile devices (I know that's not applicable atm, but still it's semantically incorrect.) I don't really have an issue with the incrementer UI, but if it really bugs you I can change it back to `text`. You just might wanna add some validation in there to make sure they're not tying alpha characters ¯\_(ツ)_/¯

[the remark this answers is not in the document]
Comment (johnhforrest) [reply src/components/modals/StakeTokensModal.vue]: What's wrong with displaying the number pad? We want them to type a number

[the remark this answers is not in the document]
Comment (sprice) [reply src/components/modals/StakeTokensModal.vue]: I'm a big fan of using markup as it's supposed to be used. I've found there to be a number of cases, especially around a11y, where experiences break when not doing this. It's easy to use CSS to remove the increment utility. Is there something else about `number` you don't like?

[the remark this answers is not in the document]
Comment (Anaphase) [reply src/components/modals/StakeTokensModal.vue]: A telephone number pad has a different connotation... and some browsers could display the result as a phone number (e.g. xxx-xxx-xxxx). I'll just change it back to `text`.

[the remark this answers is not in the document]
Comment (Anaphase) [reply src/components/modals/StakeTokensModal.vue]: 2:1 veto, leaving as `type="number"`

[the remark this answers is not in the document]
Comment (sprice) [reply src/components/modals/StakeTokensModal.vue]: Are we just giving up on `type="number"`? It's available to us for a reason 😁

[the remark this answers is not in the document]
Comment (Anaphase) [reply src/components/modals/StakeTokensModal.vue]: Sorry, I was typing my first reply before yours came in. Leaving as `number` 👍

File: src/components/modals/MetaMaskNotificationModal.vue [hunk 2]
```diff
@@ -4,17 +4,58 @@
...
+
+      <div v-if="metamaskError">
+        <p class="text-center">
+          There was a problem sending your transaction:
+        </p>
+        <pre class="text-center metamask-error">{{metamaskError}}</pre>
+      </div>
+
+      <div v-else>
+        <div v-if="currentStep === 1">
```
Comment (sprice) [inline src/components/modals/MetaMaskNotificationModal.vue]: Not a big deal but it might help to make these step values constants.

Comment (Anaphase) [reply src/components/modals/MetaMaskNotificationModal.vue]: Yeah, only problem with that is it's harder to say "go to the next step", you then have to say "go to the step with this exact name". I dunno, not opposed to that but we can revisit it later if it becomes cumbersome in the future.

**Comment by sprice** [comment]
This works great.

I'm reminded again about us soon changing a lot of this so that it's clear transactions are handled by many different tools (Mist, Toshi etc) but that's for another day.

**Pull request merged**